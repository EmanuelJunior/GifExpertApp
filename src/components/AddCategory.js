import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // String

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = ( event ) => {
        event.preventDefault();

        if( inputValue.trim().length >= 2 ){
            setCategories( categories => [inputValue, ...categories] );
            setInputValue('');
            console.log('Llegamos hasta aca!');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue }
                // onChange es parecido a el evento del teclado keydown
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
}

