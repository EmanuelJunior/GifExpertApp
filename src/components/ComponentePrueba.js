import { useEffect } from 'react';
import { Fragment } from 'react';

const ComponentePrueba = () => {

    const saludar = () => alert("Hello world");
    saludar();

    return (
        <Fragment>
            <h1 className="titulo" id='titulo' contentEditable={true}>Hola mundo desde Colom</h1>
            <p id="parr1">Esto es el primer parrafo</p>
            <p id="parr2">Este es el segundo parrafo</p>
            <b>Hola estoy en bold</b>

            <div className='anidado'>
                <p>Un parrafo</p>
                <h3>Un subtitulo</h3>
                <hr></hr>
                
                <input 
                    type='text' 
                    placeholder='Users'
                    className='userInput' 
                    value='Aca debe de ir el usuario'
                />
            </div>
        </Fragment>
    );
}

export default ComponentePrueba;