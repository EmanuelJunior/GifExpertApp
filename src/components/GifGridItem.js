import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    
    return (
        <div className='card animate__animated animate__bounceInRight'>
            <img src={ url } alt={ title }/>
            <p> { title } </p>
        </div>
    )
}
