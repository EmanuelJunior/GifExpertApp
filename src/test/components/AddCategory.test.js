import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';  

describe('Pruebas en el componente <AddCategory />', ()=>{

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    });

    test('Debe de mostrarse correctamente ', () => {
        
        expect( wrapper ).toMatchSnapshot();
    }); 
    
    test('Debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', { target: { value } });

        expect( wrapper.find('p').text().trim() ).toBe( value )
    });

    test('NO debe de postear la informacion con submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled()
    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
            const value = 'Hola mundo';
            
            // 1. Simular el inputChange
            const input = wrapper.find('input');
            input.simulate('change', { target: { value } });
            
            // 2. Simular el submit
            const form = wrapper.find('form');
            form.simulate('submit', { preventDefault(){} });
            
            // 3. setCategories se debe de haber llamado
            expect( setCategories ).toHaveBeenCalled();
            expect( setCategories ).toHaveBeenCalledTimes(1); // --> Con este metodo podemos saber expectar cuantas veces se ha ejecutado una funcion
            expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

            // 4. El valor del input debe de estar ''
            expect( input.prop('value') ).toBe( '' );


    });  
});

// describe('OTHER',()=>{
//     const setCategories = jest.fn();

//     test('Una prueba pequeÃ±a', () => {
//         const wrapper = shallow(<AddCategory setCategories={ setCategories }/>)

//         const valor = wrapper.find('p').text().trim()
//         const input = wrapper.find('input');
//         const value = 'Hola mundo'

//         input.simulate('change', {target: { value }})
//         console.log(input.prop('value'))

//         expect( wrapper ).toMatchSnapshot();
//     })
    
// })