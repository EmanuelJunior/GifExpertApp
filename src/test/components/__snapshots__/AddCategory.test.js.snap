// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> Debe de mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
